src/main/java/org/overviewproject/mime_types/MimeTypeDetector.java

```diff
@@ -136,8 +128,17 @@ private static String unwrapFutureString(CompletionStage<String> future) throws
 
         return getBytesAsync.get()
             .thenApply(bytes -> {
-                for (String magicMimeType : bytesToMimeTypes(bytes)) {
+                Iterable<String> mimeTypes = bytesToMimeTypes(bytes);
+
+                for (String magicMimeType : mimeTypes) {
                     if (globMimeTypes.isEmpty()) {
+                        // return MIME type of the superclass
```

Comment:
Hehe, good point. I can't quote the spec, because it does not specifically say how to resolve conflicts. HOWEVER, I think I found a simpler way to do the same by switching from a HashSet to a LinkedHashSet in bytesToMimeTypes(). This ensures the candidates are iterated in the order they are found and the generic superclass gets returned due to how formats are specified in the MIME file (parents first it seems), which means we achieve compatibility with the Perl MimeInfo module (https://metacpan.org/source/MICHIELB/File-MimeInfo-0.29/lib/File/MimeInfo/Magic.pm) which is nice. :)